inference-engine/src/vpu/graph_transformer/src/stages/interpolate.cpp

```diff
@@ -39,78 +60,79 @@ void FrontEnd::parseInterpolate(const Model& model, const ie::CNNLayerPtr& _laye
                                                        _layer,
-                                                       antialias,
+                                                       anti,
                                                        coordinateTransformationMode,
                                                        nearestMode,
                                                        -1.0f,
                                                        input,
                                                        output);
-            } else if (cmp(interpolateMode, "linear") || cmp(interpolateMode, "linear_onnx")) {
+            } else if (cmp(interpolateMode, g_linear) || cmp(interpolateMode, g_linear_onnx)) {
                 // current "Interp" supports modes "align_corners" and "asymmetric" only
                 // other "Interpolate" modes are translated to the default ones
-                const auto coordinateTransformationMode = _layer->GetParamAsString("coordinate_transformation_mode", "half_pixel");
+                const auto coordinateTransformation = _layer->GetParamAsString(g_coordinate_transformation_mode, g_half_pixel);
+                InterpolateCoordTransMode coordinateTransformationMode = InterpolateCoordTransMode::HalfPixel;
+                InterpolateMode mode = InterpolateMode::Linear;
+
+                if (cmp(interpolateMode, g_linear_onnx)) {
+                    mode = InterpolateMode::LinearOnnx;
+                }
+                coordinateTransformationMode = coordinateChoice(coordinateTransformation);
 
                 _stageBuilder->addInterpStage(model,
                                               _layer->name,
                                               _layer,
-                                              cmp(coordinateTransformationMode, "align_corners"),
+                                              cmp(coordinateTransformation, g_align_corners),
+                                              mode,
+                                              coordinateTransformationMode,
                                               input,
                                               output);
             } else {
-                VPU_THROW_EXCEPTION << "Current Interpolate supports 'nearest' and 'linear' modes only; layer name = " << _layer->name;
+                VPU_THROW_FORMAT("Current Interpolate supports 'nearest' and 'linear' modes only");
             }
         } else {
-            VPU_THROW_EXCEPTION << "Current Interpolate does not support paddings, batches, and resize by channels; layer name = " << _layer->name;
+            VPU_THROW_FORMAT("Current Interpolate does not support paddings, batches, and resize by channels");
         }
     } else {
-        VPU_THROW_EXCEPTION << "Current Interpolate supports (N)HWC, (N)CHW data orders only; layer name = " << _layer->name;
+        VPU_THROW_FORMAT("Current Interpolate supports (N)HWC, (N)CHW data orders only");
```

Comment:
Nitpick: Such layered code is hard to read. Is it possible to make it linear? E.g. first check input is supported (order, paddings, batches can be checked independently as separate checks, right?). Then process assuming all checks are done (in a linear fashion as well)

Reply:
Agree with Gleb
```
const auto orderIsSupported = input->desc().dimsOrder() == DimsOrder::NCHW || input->desc().dimsOrder() == DimsOrder::NHWC || input->desc().dimsOrder() == DimsOrder::CHW || input->desc().dimsOrder() == DimsOrder::HWC;
VPU_THROW_UNLESS(orderIsSupported, "Current Interpolate supports (N)HWC, (N)CHW data orders only, actual {}", input->desc().dimsOrder());

// following steps with the same indents 
```